R/get_test_data.R

```diff
@@ -42,14 +42,16 @@ get_test_data <- function(recipe, x){
     stop("insufficient training data")
   }
 
+  groups <- epi_keys(recipe)[epi_keys(recipe) != "time_value"]
+
   test_data <- x %>%
     dplyr::filter(
       dplyr::if_any(
         .cols = recipe$term_info$variable[which(recipe$var_info$role == 'raw')],
         .fns = ~ !is.na(.x)
       )
     ) %>%
-    epiprocess::group_by(geo_value) %>%
+    epiprocess::group_by(across(dplyr::all_of(groups))) %>%
```

Comment:
Looks like `across` hasn't been `@importFrom`'d, so we need `dplyr::` as well.  If you run `devtools::check()` or `devtools::check(vignettes=FALSE)` you will see
```
   get_test_data: no visible global function definition for ‘across’
```

CI should be automatically flagging this + a lot of pre-existing issues; maybe we haven't triggered it yet since it'd been set up, or maybe there's a configuration issue or something deliberately disabled.

Reply:
The reason why `check` is complaining is because this looks like a situation where
- if the user doesn't have a package attached with `across`, this would be an error
- if the user has a package attached with a different `across` function than we wanted, or defined their own `across` function as a global, this might have different behavior than wanted
This is because when a package doesn't import or use `::` on all its dependencies, R will fall back on whatever the user has defined globally or is available in attached packages.

Why does the test still pass, even if we try defining `across=function(...) stop("using a different across than intended")`?  

Short answer: because `dplyr` saves us.

Long answer: there are a few things that each would prevent some/all issues:
- `dplyr` saves us: it appears to add `across` as part of `.data` (within `group_by`), which is searched before `.env` which would contain the custom `across` (we'll find an `across` function, and it's `dplyr`'s)
- the test file `library`s in (loads and attaches) `dplyr` (we'll find an `across`, the right one unless we define our own `across` or attach another package with a different `across` --- e.g., we would get "using a different across than intended")
- the `recipes` dependency has `Depends: dplyr` for some reason, which attaches `dplyr` (we'll find an `across`, but not necessarily the right one --- this alone wouldn't prevent the custom-`ahead` issue --- e.g., we would get "using a different across than intended")

We still want to add the scope or `@importFrom` to:
- satisfy the package checks
- hedge against `dplyr` removing `across` from `.data`

Reply:
Editing the above to fix some notes.